lib/dangermattic/plugins/manifest_pr_checker.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check if the Gemfile.lock was updated when changing the Gemfile in a PR.
+  class ManifestPRChecker < Plugin
+    # Check if the `Gemfile` file was modified without a corresponding `Gemfile.lock` update
+    def check_gemfile_lock_updated
+      check_manifest_lock_updated(
+        file_name: 'Gemfile',
+        lock_file_name: 'Gemfile.lock',
+        instruction: 'Please run `bundle install` or `bundle update <updated_gem>`'
+      )
+    end
+
+    # Check if the `Podfile` file was modified without a corresponding `Podfile.lock` update
+    def check_podfile_lock_updated
+      check_manifest_lock_updated(
+        file_name: 'Podfile',
+        lock_file_name: 'Podfile.lock',
+        instruction: 'Please run `bundle exec pod install`'
+      )
+    end
```

Comment:
While this covers the most common case, there are some repositories which also contains `Podfile`+`Podfile.lock` files that are **not** at the root of the repo.

One example that comes to mind is Tumblr-iOS and its `Components/` sub-projects all hosted in the Tumblr-iOS monorepo.
Another example of such repo would be library (pod) repos which contains a Sample project, where the `MyLib.podspec` would be at the root but there would be a `SampleApp/Podfile` and `SampleApp/Podfile.lock` files there.

Given this, maybe the `check_manifest_lock_updated` helper method should not just check for that exact path at the repo root when looking at `git.modified_files`, but also consider any `git.modified_files` whose paths ends with the `file_name` (and compare it with the file that lives in the same `dirname` but has `lock_file_name` as its `basename`)?

Reply:
Hmm, indeed it might happen. I've slightly changed the code to cater for that.